backends/xboxone/xbox-one-utils.js

```diff
@@ -0,0 +1,373 @@
+ * @param {string} password The password to authenticate to the Device Portal.
+ * @return {!Promise.<Buffer>}
+ */
+async function takeScreenshot(xboxOneAddress, username, password) {
+  // https://docs.microsoft.com/en-us/windows/uwp/xbox-apps/wdp-media-capture-api
+  const {data} = await httpRequestHelper(
+      xboxOneAddress, username, password,
+      'GET', 'ext/screenshot?download=true&hdr=false');
+  return data;
+}
+
+
+/**
+ * A helper function to make HTTP requests to the Device Portal APIs on the Xbox
+ * One device.
+ *
+ * @param {string} xboxOneAddress The IP or hostname of the Xbox One device.
+ * @param {string} username The username to authenticate to the Device Portal.
+ * @param {string} password The password to authenticate to the Device Portal.
+ * @param {string} method The HTTP request method to use.
+ * @param {string} path The HTTP request path to use.
+ * @param {?FormData} formData Optional multipart form data for file uploads.
+ * @return {!Promise<{statusCode: number, data: Buffer}>}
+ */
+function httpRequestHelper(
+    xboxOneAddress, username, password, method, path, formData) {
+  return new Promise((resolve, reject) => {
+    const url = `https://${xboxOneAddress}/${path}`;
+    const options = {
+      defaultPort,  // if not specified in xboxOneAddress
+      rejectUnauthorized: false,  // do not validate HTTPS cert
+      method,
+    };
+    if (formData) {
+      options.headers = formData.getHeaders();
+    } else {
+      options.headers = {};
+    }
+
+    // NOTE: auto- prefix disables requirements around CSRF tokens, to allow
+    // automation on this API.
+    // See https://docs.microsoft.com/en-us/windows/uwp/debug-test-perf/device-portal#csrf-protection-and-scripting
+    const basicAuth = Buffer.from('auto-' + username + ':' + password, 'utf8');
+    options.headers.Authorization = 'Basic ' + basicAuth.toString('base64');
+
+    const req = https.request(url, options, (response) => {
+      const chunks = [];
+      response.on('data', (chunk) => {
+        chunks.push(chunk);
+      });
+      response.on('end', () => {
+        const statusCode = response.statusCode;
+        const data = Buffer.concat(chunks);
+        if (statusCode < 200 || statusCode > 299) {
+          const error = new Error(
+              `${method} ${url} => ` +
+              `HTTP ${statusCode}: ${data.toString('utf8')}`);
+          error.method = method;
+          error.url = url;
+          error.statusCode = statusCode;
+          error.body = data.toString('utf8');
+          reject(error);
+        } else {
+          resolve({statusCode, data});
+        }
+      });
+    });
+
+    req.on('error', reject);
+
+    if (formData) {
+      formData.pipe(req);
+    } else {
+      req.end();
+    }
+  });
+}
+
+/**
+ * Try to find MSBuild.exe from Visual Studio, to simplify configuration.
+ *
+ * @return {?string} The path to MSBuild.exe, or null if not found.
+ */
+function findMsbuild() {
```

Comment:
I tried your suggestion just now, and it doesn't work.

I ran the registry query, found the full path to MSBuild.exe within that, and used the --msbuild option in the CLI tool to do a build with that.

The copy of MSBuild found in that registry query seems to be specific to .NET, and failed to build this UWP project.  The copy found in the Visual Studio path below works, though.

Reply:
In that case, I'd suggest putting `2019` first in the list so we favor newer versions.  Also, for me, the 2017 version doesn't have a `Current` folder, and instead it is in `15.0` (referring to VS version number).

Reply:
Actually, since I can't test on VS 2017 or 2015, I think it might be best to just hard-code 2019 in this method for now.  I'll update the docs to make it clear that 2017 and 2015 should still be able to build the UWP app, but that users will have to configure the path explicitly for those.